middleware/file/zone.go

```diff
@@ -24,6 +24,8 @@ type Zone struct {
 	*tree.Tree
 	Apex Apex
 
+	dnames []*dns.DNAME
```

Comment:
I don't think we should store these separately, this, for one, means the axfr does not include them. As it iterates over Tree.

We should just set a bit, that we have DNAMEs in the zone. Ideally the normal lookup mechanism should be enough to make this work, but I'll have to double check what the RFC actually says about this... something like this?:

If there is a dname found during the lookup, we should synthesize the cnames and return those (*and*) perform additional section processing for it.